CMSgov/cms-carts-seds
# Reassign kafka event source mapping to lambda

### Description

CARTS listens to a kafka topic with SEDS data and saves it to a database to display within the CARTS report.
The old way of doing this involved having an ECS service constantly running and polling every minute for new information. This service was configured by one lambda, and sent its data to another lambda to process and save.
This PR simplifies this flow such that kafka triggers the lambda directly. We can remove all the in between infrastructure, which will save our client some money as well!

![new kafka flow(2)](https://github.com/user-attachments/assets/b47c8b72-3ccf-4154-a6b9-3d68a1490c83)


### Related ticket(s)


CMDCT-3837

---
### How to test

Normally the carts-bigmac-streams service does not deploy in feature branches. I overrode that here so we could test this flow before merging to main. We want to compare functionality in this branch to main.

Prep:
- Go to the AWS console for CARTS dev in browser
- Open these cloudwatch log groups in separate tabs
  - `/aws/lambda/carts-bigmac-streams-main-sinkEnrollmentCounts`
  - `/aws/lambda/carts-bigmac-streams-kafka-test-sinkEnrollmentCounts`
- Open these dynamodb tables in separate tabs
  - `main-stg-enrollment-counts`
  - `kafka-test-stg-enrollment-counts`
- View the `enrollmentCount` column for AL-2023 records
  - make note of each value
- Navigate to https://d2w1h2xahy1f0c.cloudfront.net/
- Log in as stateuser2@test.com
- Enter AL 2023 and go to Section 2
- Note the values in the first table under "Number of children enrolled in FFY 2023" match the db

Trigger the event:
- Go to [SEDS dev](https://mdctsedsdev.cms.gov/)
- Log in as stateuser@test.com
- Choose 2023 quarter 4
- Enter either form "21E" or "64.21E"
- Enter some values in any table for any age and hit "Save"

Verify:
- Go back to the log groups and look for the latest logs
  - If a recent log was registered, they may exist within the latest
  - The kafka-test logs may have a 30 second delay as per how the event is configured
- Verify the processing logs show up in each
  - They will say "INFO Sink message received" and then have two objects `NewImage` and `OldImage`
  - Compare the `last_modified` times and the `enrollmentCounts` values
- Refresh the databases and check the `enrollmentCount` column to see that the new value matches
- Navigate to https://d2w1h2xahy1f0c.cloudfront.net/ (or refresh if open)
- Log in as stateuser2@test.com
- Enter AL 2023 and go to Section 2
- Verify table data updates as expected ("Number of children enrolled in FFY 2023" in first table)

### Notes


Deployed this branch from main and merged `cmdct-3837` into it to simulate merging the changes to main

I already added parameters `/configuration/default/bigmac/bootstrapBroker_` to val and production envs

I know the kafka event can be configured directly on the lambda function definition, though there was an odd error with attaching the security group to the event where it would fill in as an object instead of a string. I tried so many things and could not get it to work, so I made the event mapping in the resources section.

The new lambda event has a consumer group id that is `carts-${env}`. This means that each env will only read events once and it will pick up where it left off when there are new events.

---
### Pre-review checklist


---
### Pre-merge checklist


#### Security
Changes rely on existing infrastructure and connected accounts. No new security concerns.

---
